collections/_tutorials/2022-09-12-tutorial-cvae.md

```diff
@@ -0,0 +1,296 @@
+
+$$p(z/x) = \mathcal{N}\left(k(x,y),l(x,y)\right)$$
+
+>As we will see later, minimizing the KL divergence between the approximation $$q(z/x,y)$$ and the target $$p(z/x,y)$$ also leads to finding the optimal $$k^*$$ and $$l^*$$ functions.
+
+&nbsp;
+
+### Formulation of the KL divergence
+
+Let's now reformulate the KL divergence expression
+
+$$D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right) = - \int{q(z/x,y) \cdot log\left(\frac{p(z/x,y)}{q(z/x,y)}\right) \,dz}$$
+
+Using the following ***conditional probability*** relations:
+
+$$
+p(x,y,z) = \left\{
+  \begin{array}
+    pp(y,z/x) \cdot p(x) \\
+    p(z/x,y) \cdot p(x,y)
+  \end{array}
+  \right.
+$$
+
+$$p(x,y) = p(y/x) \cdot p(x)$$
+
+the next equations can be easily obtained
+
+$$p(z/x,y) = \frac{p(y,z/x) \cdot p(x)}{p(x,y)}$$
+
+$$p(z/x,y) = \frac{p(y,z/x) \cdot p(x)}{p(y/x) \cdot p(x)}$$
+
+$$p(z/x,y) = \frac{p(y,z/x)}{p(y/x)}$$
+
+&nbsp;
+
+The previous KL divergence expression can thus be rewritten as
+
+$$D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right) = - \int{q(z/x,y) \cdot log\left(\frac{p(y,z/x)}{p(y/x) \cdot q(z/x,y)}\right) \,dz}$$
+
+$$ = - \int{q(z/x,y) \cdot 
+\left[log\left(\frac{p(y,z/x)}{q(z/x,y)}\right) + log\left(\frac{1}{p(y/x)}\right) \right]\,dz}$$
+
+$$ = - \int{q(z/x,y) \cdot 
+log\left(\frac{p(y,z/x)}{q(z/x,y)}\right)\,dz} + log\left(p(y/x)\right) \cdot \underbrace{\int{q(z/x,y)\,dz}}_{=1}$$
+
+$$D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right) \,+\, \mathcal{L} \,=\, log\left(p(y/x)\right)$$
+
+where $$\mathcal{L}$$ is defined as the ***Evidence Lower BOund (ELBO)*** whose expression is given by:
+
+$$\mathcal{L} = \int{q(z/x,y) \cdot log\left(\frac{p(y,z/x)}{q(z/x,y)}\right) \,dz}$$
+
+&nbsp;
+
+### Evidence lower bound
+
+Let's take a closer look at the previous derived equation:
+
+$$D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right) \,+\, \mathcal{L} \,=\, log\left(p(y/x)\right)$$
+
+The following observations can be made:
+* since $$0\leq p(y/x) \leq 1$$, $$log\left(p(y/x)\right) \leq 0$$
+
+* since $$x$$ and $$y$$ are known, $$log\left(p(y/x)\right)$$ is a fixed value
+
+* by definition $$D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right) \geq 0$$
+
+* since $$\mathcal{L} = -D_{KL}\left(q(z/x,y) \parallel p(y,z/x)\right)$$, $$\mathcal{L} \leq 0$$
+
+
+The previous expression can thus be rewritten as follows:
+
+$$\underbrace{D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right)}_{\geq 0} \,+\, \underbrace{\mathcal{L}}_{\leq 0} \,=\, \underbrace{log\left(p(y/x)\right)}_{\leq 0 \,\, \text{and fixed}}$$
+
+>Thus, by tweaking q(z/x,y), we can seek to maximize the ELBO $$\mathcal{L}$$, which will imply the minimization of the KL divergence $$D_{KL}\left(q(z/x,y) \parallel p(z/x,y) \right)$$, and thus to find a distribution $$q(z/x,y)$$ that is close to $$p(z/x,y)$$.
+
+&nbsp;
+
+### ELBO reformulation
+
+The ELBO $$\mathcal{L}$$ should be reformulated so to justify the loss involved in the conditional VAE framework. The corresponding derivation is provided below.
```

Comment:
_so to justify_: There's a word missing here, but I'm not able to figure out what exactly.

Reply:
Done

Reply:
New sentence: The ELBO $$\mathcal{L}$$ must be reformulated to justify the loss involved in the conditional VAE framework.